champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/bow/NapalmArrow.java

```diff
@@ -0,0 +1,267 @@
+        return Role.RANGER;
+    }
+
+    @Override
+    public SkillType getType() {
+        return SkillType.BOW;
+    }
+
+    private void doNapalm(Player player, Location arrowLocation, int level) {
+        World world = arrowLocation.getWorld();
+        List<Item> fireItems = new ArrayList<>();
+
+        for (int i = 0; i < numFlames; i++) {
+            Item fire = world.dropItem(arrowLocation.add(0.0D, 0.0D, 0.0D), new ItemStack(Material.BLAZE_POWDER));
+            ThrowableItem throwableItem = new ThrowableItem(this, fire, player, getName(), (long) (getDuration(level) * 1000L));
+            championsManager.getThrowables().addThrowable(throwableItem);
+
+            double x = (random.nextDouble() - 0.5) * velocityMultiplier;
+            double y = random.nextDouble() * yComponentVelocityMultiplier;
+            double z = (random.nextDouble() - 0.5) * velocityMultiplier;
+            fire.setVelocity(new Vector(x, y, z));
+
+            fireItems.add(fire);
+        }
+
+        new BukkitRunnable() {
+            @Override
+            public void run() {
+                if (fireItems.isEmpty()) {
+                    this.cancel();
+                    return;
+                }
+
+                fireItems.removeIf(item -> item.isDead() || !item.isValid());
+
+                if (!fireItems.isEmpty()) {
+                    Item randomFire = fireItems.get(random.nextInt(fireItems.size()));
+                    world.spawnParticle(Particle.LAVA, randomFire.getLocation(), 0);
+                }
+            }
+        }.runTaskTimer(champions, 0L, 1L);
+    }
+
+    @Override
+    public void onThrowableHit(ThrowableItem throwableItem, LivingEntity thrower, LivingEntity hit) {
+        if (hit instanceof ArmorStand) {
+            return;
+        }
+
+        if (hit.equals(thrower)) {
+            return;
+        }
+
+        if (thrower instanceof Player damager) {
+            int level = getLevel(damager);
+            hit.setFireTicks((int) (getBurnDuration(level) * 20));
+
+            CustomDamageEvent cde = new CustomDamageEvent(hit, damager, null, EntityDamageEvent.DamageCause.CUSTOM, getDamage(level), false, "Napalm");
+            cde.setDamageDelay(damageDelay);
+            UtilDamage.doCustomDamage(cde);
+        }
+    }
+
+    @Override
+    public void activate(Player player, int level) {
+        player.getWorld().playSound(player.getLocation(), Sound.ENTITY_BLAZE_AMBIENT, 2.5F, 2.0F);
+        active.add(player.getUniqueId());
+    }
+
+    @Override
+    public void onHit(Player damager, LivingEntity target, int level) {
+        //ignore
+    }
+
+    @EventHandler
+    public void onArrowDamage(CustomDamageEvent event){
+        if (!(event.getDamager() instanceof Player player)) return;
+        if (!(event.getProjectile() instanceof Arrow arrow)) return;
+        if (!hasSkill(player)) return;
+        if (!napalmArrows.containsValue(arrow)) return;
+
+        int level = getLevel(player);
```

Comment:
Couple of issues here,

since you are getting the level, you dont need hasSkill, just check if level > 0

Then napalmArrows you're doing the same thing as biotic shot, it might be better in your case to instead make it WeakHashMap<Arrow, UUID>, this way if more than 1 arrow is active from the same shooter it wont cause issues